### DISCUSSION TITLE
"Premium" costs?

### DISCUSSION THREAD

**Discussion by 04alband** [CJNE/ha-myenergi#556] [Q&A] [answered]
Hi, I'm sure someone can answer this really easily but I just wanted to confirm as I was unable to find it explicitly...
Does the MyEnergi API (or at least the extent of it which is used by this HACS integration), incur additional charges?
I ask because of this nicely vague and slightly hidden message:
![image](https://github.com/user-attachments/assets/75a1456a-8b21-4ff4-9f1e-9a8a014f5133)
[Found at the bottom of this page](https://support.myenergi.com/hc/en-gb/articles/5069627351185-How-do-I-get-an-API-key)

Like I say, it seems likely people aren't being charged otherwise I expect I would find some threads about it. But it'd be nice to know with some more certainty.

Thanks to all those who've contributed to this integration. I've just installed HACS and set it up and am really looking forward to using it :)

David

**Comment by G6EJD** [comment, marked as answer]
That was a comment made by Dr Chris Horne of MyEnergi, I know the full history, his comment and statement was made in response to excessive API calls, at rates of 1/sec. A lot of people set this rate largely because they don’t understand systems engineering and nothing in the EV or water heating systems work they fast, mainly every 10-mins. Their Amazon Web Services used for the API service provision is not low cost and Usage tariff based.

And yes, MyEnergi were planning premium API services, but they seems to have abated of late snd there has been little mention of it.

So there are no API costs, but if the abuse continues then I’d expect a call limit to be imposed, first say matching most other systems like SolarEdge who allow 300 calls per day or about every 15-sec.

I observe that MyEnergi are very rare in allowing open/unrestricted API access.

**Reply by 04alband** [reply]
Ok thanks. What you say makes total sense. And yeah, as frustratingly slow as I find the app, having a free, open API that allows control over the device is super useful. Of course would be nice if it were local than cloud, but that's off topic...

Thanks for the speedy response :)
